src/react/components/pages/editorPage/canvas.test.tsx

```diff
@@ -568,11 +568,26 @@ describe("Editor Canvas", () => {
     });
 
     it("Clears all regions from asset", async () => {
-        const wrapper = createComponent().find(Canvas);
+        const wrapper = createComponent();
         const clearConfirm = wrapper.find(Confirm) as ReactWrapper<IConfirmProps>;
         clearConfirm.props().onConfirm();
 
         await MockFactory.flushUi();
         expect(wrapper.state().currentAsset.regions).toEqual([]);
     });
+
+    it("Tags are re-applied when the project tags are updated", () => {
+        const wrapper = createComponent();
+        const assetMetadata = wrapper.props().selectedAsset;
+        const project = { ...wrapper.props().project };
+        const tags = [...project.tags];
+        tags[0].color = "#FFCCFF";
+        project.tags = tags;
+
+        const editor = wrapper.instance().editor;
+        const updateTagsSpy = jest.spyOn(editor.RM, "updateTagsById");
+
+        wrapper.setProps({ project });
+        expect(updateTagsSpy).toBeCalledTimes(assetMetadata.regions.length);
```

Comment:
I feel like this is exposing/dependent on internal implementation.  How about testing that all the regions have a new tag color?

Reply:
Our tag color is modified directly by the tags input.  This bug that the canvas component didn't update to reflect tag updates.  Extending this test beyond calling into canvas tools API would be re-testing the internal implementation of the canvas itself.